audformat/core/utils.py

```diff
@@ -286,6 +286,95 @@ def intersect(
     return index
 
 
+def join_labels(
+        labels: typing.Sequence[typing.Union[typing.List, typing.Dict]],
+):
+    r"""Combine scheme labels.
+
+    This might be helpful,
+    if you would like to combine two databases
+    that have the same scheme,
+    but with different labels:
+
+    .. code-block:: python
+
+        labels = audformat.utils.join_labels(
+            [
+                db.schemes['scheme'].labels,
+                db_new.schemes['scheme'].labels,
+            ]
+        )
+        db.schemes['scheme'].replace_labels(labels)
+        db_new.schemes['scheme'].replace_labels(labels)
+        db.update(db_new)
+
+    Args:
+        labels: sequence of labels to join
+
+    Returns:
+        joined labels
+
+    Raises:
+        ValueError: if labels argument is a dict
+        ValueError: if labels are of different type
+        ValueError: if the labels are dicts
+            and contain different values for the same key
+        RuntimeError: if label type is not ``list`` or ``dict``
+
+    Example:
+        >>> join_labels([{'a': 0, 'b': 1}, {'b': 1, 'c': 2}])
+        {'a': 0, 'b': 1, 'c': 2}
+
+    """
+    if isinstance(labels, dict):
+        raise ValueError(
+            "'labels' has to be a sequence, yours is a 'dict'"
+        )
+
+    if len(labels) == 0 or isinstance(labels, dict):
+        return labels
+
+    if not isinstance(labels, list):
```

Comment:
Again, why do we need this when we expect a list as input?

Reply:
We state in the docs that we accept a sequence as input, so it might be that we get tuple, which we need to convert to a list at this stage.

Reply:
But are you sure it's necessary? Or is it the `audeer.flatten_list()` that requires a `list` as input?

Reply:
At the moment yes. Before I tried to implement a version without `audeer.flatten_list()`, but there I also got an error for tuples.
So, I haven't found a better way for now. As we want to return always a list we have to convert at some point anyway.

Reply:
It's in general not very pretty the code for joining the labels. I guess there must be an easier solution.

Reply:
It's in general not very pretty the code for joining the labels. I guess there must be an easier solution.